### PR TITLE
add stateful_object solution

### PR DESCRIPTION
**Pull request by Serhii-Dr**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,25 @@
...
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => delete state[key]);
+        break;
+      case 'clear':
+        if (Object.keys(state).length > 0) {
```
Comment (Viktor-Kost) [inline src/transformState.js]: Checking the length of Object.keys(state) is redundant. The forEach loop will not execute if there are no keys, so this check can be removed for cleaner code.

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,25 @@
...
+          Object.keys(state).forEach(key => delete state[key]);
+        }
+        break;
+      default:
+        return state;
```
Comment (Viktor-Kost) [inline src/transformState.js:20]: In the 'default' case of switch, it is not necessary to return 'state'. Switch statement doesn't require a default return as function will return 'state' at the end regardless.

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/transformState.js (hunk 2 above)
Comment (VitaliyBondarenko1982) [reply src/transformState.js:20]: https://github.com/mate-academy/js_stateful-object/pull/2537/files#r1252276239
not fixed this comment. it means that no need to return state. better to throw some error for example.

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Looks good

File: src/transformState.js (hunk 2 above)
Comment (Serhii-Dr) [reply src/transformState.js:20]: I think i'm not 100% understood this issue
should I just in 'default' case to throw 'new Error' object and not to return state both from switch statement and function at all?

**Pull request closed without merging**